smartsim/wlm/__init__.py

```diff
@@ -39,28 +39,30 @@ def detect_launcher() -> str:
     # Precedence: PBS, Cobalt, LSF, Slurm, local
     if which("qsub") and which("qstat") and which("qdel"):
         qsub_version = run(
-            ["qsub", "--version"], shell=False, capture_output=True, encoding="utf-8"
+            ["qsub", "--version"],
+            shell=False,
+            capture_output=True,
+            encoding="utf-8",
+            check=False,
         )
         if "pbs" in (qsub_version.stdout).lower():
             return "pbs"
         if "cobalt" in (qsub_version.stdout).lower():
             return "cobalt"
-    if (
-        which("bsub")
-        and which("jsrun")
-        and which("jslist")
-        and which("bjobs")
-        and which("bkill")
+    if all(
+        [which("bsub"), which("jsrun"), which("jslist"), which("bjobs"), which("bkill")]
```

Comment:
Are the [] needed? You've removed them from many other places

Reply:
Applies here and to the next block

Reply:
all takes an iterable. without parens or brackets, i'd be giving it multiple arguments instead.